extensions/sockets/src/main/java/org/mule/module/socket/internal/AbstractSocketProperties.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.internal;
+
+import org.mule.module.socket.api.SocketProperties;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.ConfigName;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+public abstract class AbstractSocketProperties implements SocketProperties
+{
+
+    public static final int DEFAULT_BUFFER_SIZE = 16384; // 1024 * 16
+    /**
+     * The name of this config object, so that it can be referenced by config elements.
+     */
+    @ConfigName
+    protected String name;
+
+    /**
+     * The size of the buffer (in bytes) used when sending data, set on the socket itself.
+     */
+    @Parameter
+    @Optional(defaultValue = "16384")
```

Comment:
can you have a private static constant for this which is a String.valueOf() the int one?